src/primitives/torus.jl

```diff
@@ -0,0 +1,54 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Torus(majorRadius, minorRadius)
+
+A torus with `majorRadius` and `minorRadius` with the z axis as its axis of 
+rotation and the xy plane as its plane of reflection. 
+
+"""
+struct Torus{T} <: Primitive{3,T}
+  majorRadius::T
+  minorRadius::T
+end
+
+function Torus(majorRadius::T1, minorRadius::T2) where {T1,T2}
+  T = promote(T1, T2)  
+  Torus(T(majorRadius), T(minorRadius))
+end
+
+paramdim(::Type{<:Torus}) = 2
+
+isconvex(::Type{<:Torus}) = false
+
+isperiodic(::Type{<:Torus}) = (true, true)
+
+minorRadius(t::Torus) = t.minorRadius
+
+majorRadius(t::Torus) = t.majorRadius
+
+# https://en.wikipedia.org/wiki/Torus
+function measure(t::Torus)
+  R, r = t.majorRadius, t.minorRadius
+  2π^2 * R * r^2
+end
+
+Base.length(t::Torus) = measure(t)
+
+volume(t::Torus) = measure(t)
```

Comment:
a torus doesn't have a volume
```suggestion
```

Reply:
What is your rationale here? A torus can mean the surface or the volume bounded by this surface.

Reply:
The mathematical definition of a Torus is that of a surface: https://en.wikipedia.org/wiki/Torus

We only provide the alias `volume` for geometries with parametric dimension 3. The torus has parametric dimension 2 as a surface, hence only `area` should be introduced as an alias to `measure`.